tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -1029,3 +1029,10 @@ def find_interface(tests, id=nil, skipcheck=true)
   end
   intf
 end
+
+def create_bridge_domain(tests)
+  # BDI_non_default test case requires to create a bridge domain
+  return unless platform == 'n7k'
+  bd = tests[:bdi_name][/(\d+)/]
+  config_bridge_domain(agent, bd)
+end
```

Comment:
`utilitylib.rb` is for common methods so this shouldn't be here unless you intend to use it in other test files. 
But I don't think you need it at all. Just do this in `test_interface.rb` instead:

```
+  if platform_supports_test(tests, 'BDI_non_default')
     logger.info("\n#{'-' * 60}\nSection 6. BDI Property Testing")
     bd = tests[:bdi_name][/(\d+)/]
-    create_bridge_domain(tests)
+    config_bridge_domain(agent, bd)
     test_harness_interface(tests, 'BDI_non_default')
+  end
```